torchaudio-contrib/io.py

```diff
@@ -0,0 +1,48 @@
+import torch
+import numpy as np
+import warnings
+from scipy.io import wavfile
+import librosa
+
+def load(filename, sr=44100, mono=True, norm=False, device='cpu', dtype=np.float32):
+    """
+    Generic wrapper for reading an audio file.
+    Different libraries offer different speeds for this, so this routine is the
+    'catch-all' for whatever read routine happens to work best
+
+    Tries a fast method via scipy first, reverts to slower librosa when necessary.
+    """
+    # first try to read via scipy, because it's fast
+    scipy_ok = False
+    with warnings.catch_warnings(record=True) as w:
+        warnings.simplefilter("error")    # scipy throws warnings which should be errors
+        try:
+            out_sr, signal = wavfile.read(filename)
+            scipy_ok = True
+        except wavfile.WavFileWarning:
+            print("read_audio_file: Warning raised by scipy. ",end="")
+
+    if scipy_ok:
```

Comment:
That's not a good style. I understand you're doing this to get out of the `with` block, but a better solution would be wrapping the "warnings-as-exceptions" scipy call in a separate function, or can't you just move it inside `try:`?

Reply:
This is *recommended* style [straight from the Python Docs](https://docs.python.org/3/library/warnings.html#testing-warnings).

Reply:
I meant the technique of setting a boolean flag somewhere and then checking for it later.